compiler/IREmitter/Payload/patches/vm_insnhelper.c

```diff
@@ -978,6 +979,24 @@ VALUE sorbet_vm_instance_variable_get(struct FunctionInlineCache *getCache, IVC
     return sorbet_callFuncWithCache(getCache, VM_BLOCK_HANDLER_NONE);
 }
 
+VALUE sorbet_vm_instance_variable_set(struct FunctionInlineCache *setCache, IVC varCache, rb_control_frame_t *reg_cfp,
+                                      VALUE recv, ID var, VALUE value) {
+    sorbet_vmMethodSearch(setCache, recv);
+    rb_method_definition_t *setDef = setCache->cd.cc.me->def;
+    if (setDef->type == VM_METHOD_TYPE_CFUNC &&
+        setDef->body.cfunc.func == sorbet_vm_Kernel_instance_variable_set_func()) {
```

Comment:
Is it worth wrapping this in `LIKELY`?
```suggestion
    if (LIKELY(setDef->type == VM_METHOD_TYPE_CFUNC &&
        setDef->body.cfunc.func == sorbet_vm_Kernel_instance_variable_set_func())) {
```

Reply:
Ehhh...maybe?  My sense is that all the pointer chasing involved in method lookup and these checks are likely to swamp any benefit gained from branch layout here.  I'll defer this until later; I think this and some related things really want a `sorbet_method_cfunc_is` primitive along the lines of `vm_method_cfunc_is`.